### PR TITLE
Minimax routines

### PR DESCRIPTION
**Pull request by aziziph**
The minimax routines has been added as a separate folder.

### CODE REVIEW

**Review by Panadestein — APPROVED** [review]
Everything looks ok to me. I only had a question regarding the choice of indexes on the Pade recursion formulas, but this is only curiosity.

File: GX-TimeFrequency/COS_SIN_WEIGHTS/mathconstants.f90 [hunk 1]
```diff
@@ -0,0 +1,147 @@
+!--------------------------------------------------------------------------------------------------!
+!   CP2K: A general program to perform molecular dynamics simulations                              !
```
Comment (Panadestein) [inline GX-TimeFrequency/COS_SIN_WEIGHTS/mathconstants.f90:2]: CP2K?

File: GX-TimeFrequency/COS_SIN_WEIGHTS/mp2_types.f90 [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+!   SPDX-License-Identifier: GPL-2.0-or-later                                                      !
+!--------------------------------------------------------------------------------------------------!
+
+! **************************************************************************************************
+!> \brief Types needed for MP2 calculations
+!> \par History
+!>       2011.05 created [Mauro Del Ben]
+!> \author MDB
+! **************************************************************************************************
+MODULE mp2_types
```
Comment (Panadestein) [inline GX-TimeFrequency/COS_SIN_WEIGHTS/mp2_types.f90:14]: This module should have a more appropriate name.

File: GX-TimeFrequency/COS_SIN_WEIGHTS/mp2_types.f90 [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+   USE cp_eri_mme_interface,            ONLY: cp_eri_mme_finalize,&
```
Comment (Panadestein) [inline GX-TimeFrequency/COS_SIN_WEIGHTS/mp2_types.f90:15]: The `cp` prefix is CP2K-ish, so maybe let's make it greeX-ish

**Comment by AlexBuccheri** [comment]
The analytic continuation does not need to be part of this merge. Please remove it, as I expect minimax to get merged first

- [ ] Remove pade from this branch

File: GX-TimeFrequency/COS_SIN_WEIGHTS/minimax_exp.f90 [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+         ALLOCATE (aw(2*k_mm(i_l)))
+         CALL get_minimax_coeff_low(i_l, aw)
+         error_r = get_minimax_numerical_error(Rc, aw)
+         DEALLOCATE (aw)
+         i_mm = MERGE(i_r, i_l, error_l .LE. error_r)
+      END IF
+
+   END SUBROUTINE get_best_minimax_approx_k53
+
+! **************************************************************************************************
```
Comment (AlexBuccheri) [inline GX-TimeFrequency/COS_SIN_WEIGHTS/minimax_exp.f90:237]: This is a unit test, it should not be here

**Review by AlexBuccheri — COMMENTED** [review]
This is no where near ready:

- It completely ignores the existing build system - it just happens to be occupying the same repo.
- It's far too big to rationalise what's going on. mp2_grids is alone ~900 lines
- Reason merges are 400 lines or less, total.
- There is no consistency in the formatting or spacing - I suggest using fprettify (pip install or as a VS code plug-in) to tidy it up... or even CP2K's fork of it
- And honestly, the initial code in CP2K reads like garbage in the first place 
- There is no automated testing to validate it link, runs and gives expected results

**Comment by gmatteo** [comment]
I close this PR as it's superseded by #22 that contains all the most important modifications done by Maryam

**Pull request closed without merging**